### ISSUE TITLE
Travis yml script error

### ISSUE THREAD

**Issue by ghost** [testdrivenio/testdriven-app-2.4#13]
# Continuous Integration

## Part 2, Chapter 3

on the `.travis.yml` you have the following:

    before_script:
        - docker-compose -f docker-compose-dev.yml up --build -d

this brakes Postgres connection and tests do not pass on Travis. I solved it by reversing the order on `--build` and `-d`

**Comment by travs** [comment]
@orlando-the-fit 

Hey, thanks for this, it seems to have fixed my build.

I tried to find out why this happens but I still don't understand why.
Do you have any docs on how changing the order of options makes a difference?

**Comment by ghost** [comment]
honestly not 100% sure but I guess `--build` should always be the last flag because `-d` regards `up` command and only after you should ask to build...

**Comment by travs** [comment]
Ah actually not sure that this changes anything.
My build randomly fails with or without this change, so the problem is still present.

One simple way to fix this is to wait some "reasonable" interval as part of `.travis.yml`.
Like adding `sleep 10` at the end of `before_script` or something.

Trying to find a more robust solution now; will update when I find it.

## Update

tl;dr just use `sleep`

---

I ended up trying something like this:

`waitport.sh`:
```
#!/bin/bash

PORT=$1
echo "Waiting for port $PORT"
while ! nc -z localhost $PORT; do
  sleep 0.1
  echo -n .
done
echo "Port $PORT open"
```

then waiting for the relevant ports (from our docker-compose-dev file) in `travis.yml`:
```
...
before_script:
  - docker-compose -f docker-compose-dev.yml up --build -d
  - ./waitport.sh 5435
  - ./waitport.sh 5001
  - ./waitport.sh 80
...
```

I think it still only works by delaying though, since the ports are immediately open and pass `nc -z`.
Probably more robust would be something checking the database itself, as in [this link](https://www.marksayson.com/blog/wait-until-docker-containers-initialized/), but I don't have time to look at how to install `psql` into the CI environment.

### Note

There was [also the possibility](https://github.com/peter-evans/docker-compose-healthcheck/blob/cb3bc1b59efe015f4333837c956a9038789fc8f0/README.md) to use `HEALTHCHECK` and a conditional `depends_on`, but this was removed in docker 3.0 unfortunately.
